### PR TITLE
Instance: Container shutdown edge case fixes

### PR DESCRIPTION
**Pull request by tomponline**
As part of diagnosing the evacuation test failures when `d.c.Stop()` seems to take much longer than it should (sometimes) I also made some additional changes to the container shutdown and stop processes in order to squash some edge case issues and spend less time in liblxc.

Adds support for non-expiring operation locks, and avoids the need for a go routine that runs and periodically extends the lock.

Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>

### CODE REVIEW

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please

File: lxd/api_cluster.go [hunk 1]
```diff
@@ -3305,7 +3309,7 @@ func restoreClusterMember(d *Daemon, r *http.Request) response.Response {
 
 					// Wait for the forceful stop to complete.
 					err = stopOp.Wait()
-					if err != nil {
+					if err != nil && !strings.Contains(err.Error(), "The instance is already stopped") {
```
Comment (tomponline) [inline lxd/api_cluster.go:3318]: @stgraber this is a hack, but the operation response codes don't align with HTTP response statuses, so I didn't feel like I could just introduce a 409 conflict one. Do you have any preference on how to introduce the ability to return HTTP response codes to operation responses?

**Comment by tomponline** [comment]
@stgraber the `lxc stop --help` shows:

>       --timeout               Time to wait for the instance before killing it (default -1)

But from what I can see, the `--timeout` just indicates the time before the command returns if the container doesn't shutdown in the timeout specified. Shall I update the LXD's behaviour to align with this help text or alter the help text?

**Comment by stgraber** [comment]
Probably better to fix the help message, less likely to surprise existing users.

File: lxd/api_cluster.go (hunk 1 above)
Comment (stgraber) [reply lxd/api_cluster.go:3312]: Hmm, not sure. Operations aren't directly related to REST API calls, so it makes some sense that they're not expected to return HTTP status codes. But having something better than arbitrary string matching would be nice though.
